### PR TITLE
Add some enhancements.

### PR DESCRIPTION
**Pull request by tobiasraabe**
#### Summary

- Add some pre-commit hooks.
- Add some top-level imports to import lower-level functions like ``robust_cholesky``.
- Some cleanups.

#### Todo

- [x] Remove logging information from the database since the path is stored as a os-specific path which cannot be unpickled on a different os.

### CODE REVIEW

File: docs/source/explanations/differentiation/richardson_extrapolation.rst [hunk 1]
```diff
@@ -152,15 +152,15 @@ which we write in shorthand notation as
...
+From looking at equation (:math:`\ast`) we see that an improved estimate of :math:`L`
+can be obtained by projecting :math:`g` onto :math:`H`.
 
 
 Remark
 ######
 
-To get a better intuition for (:math:`\ast`) consider :math:`H` in more
-detail.  For the sake of clarity let :math:`\theta = \phi = 2`.
+To get a better intuition for (:math:`\ast`) consider :math:`H` in more detail.  For the
```
Comment (tobiasraabe) [inline docs/source/explanations/differentiation/richardson_extrapolation.rst:162]:
```suggestion
To get a better intuition for (:math:`\ast`) consider :math:`H` in more detail. For the
```

File: MANIFEST.in [hunk 2]
```diff
@@ -15,11 +15,14 @@ include *.sh
 include *.yaml
 include *.yml
 
+exclude pytask.ini
+
 graft docs
```
Comment (tobiasraabe) [inline MANIFEST.in:20]: I think the docs should not be part of the package. Nobody is looking for them in ``site-packages/estimagic`` anyways.

Comment (janosg) [reply MANIFEST.in:20]: Will postpone until I improve packaging in general.

**Review by janosg — APPROVED** [review]
Thanks!

**Pull request merged**